ObjectiveGit/GTBranch.m

```diff
@@ -193,7 +193,12 @@ - (GTBranch *)trackingBranchWithError:(NSError **)error success:(BOOL *)success
 }
 
 - (BOOL)updateTrackingBranch:(GTBranch *)trackingBranch error:(NSError **)error {
-	int result = git_branch_set_upstream(self.reference.git_reference, trackingBranch.shortName.UTF8String);
+  int result = GIT_ENOTFOUND;
+  if (trackingBranch.branchType == GTBranchTypeRemote) {
+    result = git_branch_set_upstream(self.reference.git_reference, [trackingBranch.name stringByReplacingOccurrencesOfString:[GTBranch remoteNamePrefix] withString:@""].UTF8String);
```

Comment:
Interesting! Does it even make sense to set the upstream for a remote branch? Should we assert that it's a local branch instead?